### ISSUE TITLE
Lazy rendering.

### ISSUE THREAD

**Issue by vadimpopa** [cristijora/vue-tabs#8]
I'm beginner in VueJs. I've been trying to optimize some 2 level nested tabs with subtabs that have at least 6 highcharts each one.  Currently vue-tabs is lazy rendering, which is good for some scenarios, but thinking how to experiment if still doing lazy rendering but with render only once, so the active tab is not destroyed after transition so next time is not going to be rendered again. Any ideas ?

**Comment by cristijora** [comment]
@vadimpopa This can be achieved with https://vuejs.org/v2/api/#keep-alive

**Comment by vadimpopa** [comment]
Yes I saw that, but did a few experiments, haven't noted any difference. So according to you, in the following template where I should add it ?

```
<template>
  <div class="vue-app-wrapper">
    <vue-tabs>
      <v-tab v-for="(tab, i) in tabs" :key="i" :title="tab.name">
        <vue-tabs>
          <v-tab v-for="(subTab, j) in tab.subTabs" :key="j" :title="subTab.name">
            <div v-for="(type, index) in subTab.charts" :key="index">
              <chart class="Grid-cell" :query="tab.searchQuery" :type="type"></chart>
            </div>
          </v-tab>
        </vue-tabs>
      </v-tab>
    </vue-tabs>
  </div>
</template>
```

**Comment by cristijora** [comment]
Let's create a jsfiddle and start from there. Please Create one with your example and maybe put something else than charts and point what you want to cache.

**Comment by vadimpopa** [comment]
I'll work on it as soon as I'll get more time to play. I noticed that keep-alive is not working well with the v-for, especially in such tabs context. Caching worked but it rendered only the first tab.

**Comment by cristijora** [comment]
Let's start from here https://jsfiddle.net/b44cc4dq/87/

**Comment by vadimpopa** [comment]
Ok, see the screenshot, this is what I've got.

<img width="792" alt="screen shot 2017-06-13 at 2 45 08 pm" src="https://user-images.githubusercontent.com/3592131/27080979-07663866-5047-11e7-9ba8-816fc09fad2d.png">

**Comment by cristijora** [comment]
Maybe I'm missing something but tabs are already kept in the DOM. Internally they use v-show to display content which sets `display:none` on the tab.
You can see that all tabs are present in the DOM but only one is visible. The rest have `aria-hidden="true"`
This also applies to sub-tabs so you might not need keep-alive at all. So 

> so the active tab is not destroyed after transition so next time is not going to be rendered again
Is actually true by default since tabs are rendered only once and they are present in the DOM 

![image](https://user-images.githubusercontent.com/15955045/27081995-c20bb7e6-504b-11e7-867a-f16cf7af04d8.png)

**Comment by vadimpopa** [comment]
Yep, I you missed pushing a new release with latest changes, and I missed as well this but only now noticed. The npm version is from 9 May, but that fiddle is working with the latests which has the 10 days ago changes. The npm version is working like with v-if.

**Comment by cristijora** [comment]
Oh ok. Thanks for pointing that out. I will try to do small releases with like 1-2 issues from now on so I don't encounter such situations later on. For now you can point to a github repo in package.json so you get the latest version or wait for the next release in 1-2 days.

**Comment by vadimpopa** [comment]
Nah, since yours is not going to have lazy rendering it's not suitable anymore for me. I've tested a bunch of libraries like vuetify , vue-material , only vuetify has lazy rendering on tabs (v-if instead of v-show). But besides lazy rendering I need the "render only once"/deferred rendering feature: so when the tab first is activated the content is rendered, next time when the tab is activated again the content should be in DOM already. This called "deferred rendering", ExtJS has this and works very well. I may need to implement the tabs myself with this feature.

**Comment by cristijora** [comment]
Sure, putem vb joracristi@gmail.com. I will think over the issue and see if I can find some solution for that

**Comment by cristijora** [comment]
Ai incercat sa te uiti peste https://vuejs.org/v2/api/#v-once ?

**Comment by vadimpopa** [comment]
I have v-once, but is not helping much....anyways looks like vuikit (https://github.com/vuikit/vuikit) does deferred rendering, I haven't looked much into, just did a quick tests.

**Comment by SSmJaE** [comment]
```ts
    @Prop({ default: true }) keepAlive!: boolean;
    @Prop({ default: true }) lazy!: boolean;
    loaded = false;

    render() {
        let loadContentFlag: boolean = false;
        let needToFigureIsActiveAndKeepAlive: boolean = false;

        if (this.lazy) {
            if (this.loaded) {
                needToFigureIsActiveAndKeepAlive = true;
            }
        } else {
            needToFigureIsActiveAndKeepAlive = true;
        }

        if (needToFigureIsActiveAndKeepAlive) {
            if (this.active) {
                loadContentFlag = true;
            } else {
                if (this.keepAlive) {
                    loadContentFlag = true;
                }
            }
        }

        return (
            <section
                class="tab-container"
                id={`p-${this.tabId}`}
                aria-labelledby={`t-${this.tabId}`}
                role="tabpanel"
                v-show={this.active}
            >
                   {loadContentFlag && this.$slots.default}
            </section>
        );
    }
```
